### PR TITLE
Ruby Basics covered

### PR DESCRIPTION
**Pull request by abp437**
Note the Conventions followed and the Comments also.
The Inline Comments are basically my understanding of concepts.

Object-oriented will be covered in another PR.

### CODE REVIEW

File: ruby-basics/oop.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  # 'to_s' method is defined in Object class and hence all ruby objects have method 'to_s'.
+  # For example when we do string interpolation then 'to_s' method is called. puts invokes 'to_s' method too.
+  def to_s
+    "#{email}: $#{total}"
+  end
+
+end
+
+orders = []
+1.upto(5) { |n| orders << Order.new('abp437@gmail.com',"#{n*10}") }
```
Comment (shirishparsekar) [inline ruby-basics/oop.rb]: 1. space after comma (a convention)
2. no need of interpolation for n*10

File: ruby-basics/practice.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Everything you manipulate in Ruby is an object, and the results of those manipulations are themselves objects.
+
+# Function starts with the 'def' keyword.
+# Function blocks end with the 'end' keyword.
+# No need to declare variables explicitly, eg. 'result' it'll figure itself out.
+def say_hello(name)
+  result = "Hello #{name}"
+  return result
```
Comment (shirishparsekar) [inline ruby-basics/practice.rb]: Ruby implicitly returns the result of last executed statement,
so explicit return is not required in this case.
Also, you don't need to store the string here.
You can simply say 
```
def say_hello(name)
  "Hello #{name}"
end
```

File: ruby-basics/practice.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+end
+# The default value returned by a Ruby method is the value of the last expression evaluated.
+# So in the above method we should have also skipped 'return result' statement.
+
+# Each object in Ruby has a unique identifier that is the Object Id.
+# Also new instances are created with the 'new' keyword.
+
+# Single quotes for plain strings.
+puts (say_hello('Akshay'))
```
Comment (shirishparsekar) [inline ruby-basics/practice.rb]: as a convention, we don't use parenthesis while calling a built in method

File: ruby-basics/practice.rb [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+# Function starts with the 'def' keyword.
+# Function blocks end with the 'end' keyword.
+# No need to declare variables explicitly, eg. 'result' it'll figure itself out.
+def say_hello(name)
+  result = "Hello #{name}"
+  return result
+end
+# The default value returned by a Ruby method is the value of the last expression evaluated.
+# So in the above method we should have also skipped 'return result' statement.
```
Comment (shirishparsekar) [inline ruby-basics/practice.rb]: :+1:

File: ruby-basics/practice.rb (hunk 3 above)
Comment (abp437) [reply ruby-basics/practice.rb]: I had read that parenthesis isn't necessary for a single built-in function, but when chaining functions then parenthesis are useful. Is it correct?

File: ruby-basics/practice.rb [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+# Each object in Ruby has a unique identifier that is the Object Id.
+# Also new instances are created with the 'new' keyword.
+
+# Single quotes for plain strings.
+puts (say_hello('Akshay'))
+# Double quotes for Interpolated/Formatted Strings, single quotes won't work in such cases.
+puts "Hello \nWorld"
+
+# Naming conventions:
```
Comment (abp437) [inline ruby-basics/practice.rb]: Are all the below-stated naming conventions correct?

File: ruby-basics/practice.rb (hunk 3 above)
Comment (shirishparsekar) [reply ruby-basics/practice.rb]: Yes, that's right.
But this is not chaining.
Chaining will be when you'll say something like
`metho_call_1(param).method_call_2(param)`

File: ruby-basics/practice.rb (hunk 5 above)
Comment (shirishparsekar) [reply ruby-basics/practice.rb]: > Local variables, method names and params - Lowercase separated by underscores.
This is correct

> Class names, module names and constants - Starts with uppercase and multi word follows MixedCase.
Class names and module names => `Student`, `SpecialStudent`
constants => `PAGE_SIZE = 10`

File: ruby-basics/oop.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  # 'to_s' method is defined in Object class and hence all ruby objects have method 'to_s'.
+  # For example when we do string interpolation then 'to_s' method is called. puts invokes 'to_s' method too.
+  def to_s
+    "#{email}: $#{total}"
+  end
+
+end
+
+orders = []
+1.upto(5) { |n| orders << Order.new('abp437@gmail.com', n*10) }
```
Comment (shirishparsekar) [inline ruby-basics/oop.rb]: Another thing over here, as a convention, we leave a space around operators

**Comment by shirishparsekar** [comment]
For all the community standard conventions, you can refer this
https://github.com/rubocop-hq/ruby-style-guide

**Pull request merged**